File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.py

from itertools import combinations
lst = []

for i in range(9):
    n = int(input())
    lst.append(n)

lst.sort()

new_sum = sum(lst)
ans = new_sum - 100

com_lst = list(combinations(lst, 2))

for i in com_lst:
    if sum(i) == ans:
        lst.remove(i[0])
        lst.remove(i[1])
        break

for i in lst:
    print(i)

File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.java
import java.util.*;

public class Main {

	public static void main(String[] args) {

		List<Integer> lst = new ArrayList<>();
		Scanner sc = new Scanner(System.in);

		int sum = 0;
		for (int i = 0; i < 9; i++) {
			int cm = sc.nextInt();
			lst.add(cm);
			sum += cm;
		}
		
		int a = 0;
		int b = 0;
		int minus = sum - 100;
		
		System.out.println();

		xx: for (int i = 0; i < 8; i++) {
			for (int j = i + 1; j < 9; j++) {

				if (lst.get(i) + lst.get(j) == minus) {
					a = i;
					b = j;
					break xx;
				}

			}
		}

		lst.remove(b);
		lst.remove(a);

		Collections.sort(lst);
		
		for (Integer integer : lst) {
			System.out.println(integer);
		}
	}

}
